packages/transaction-controller/src/utils/gas-fees.ts

```diff
@@ -67,12 +95,26 @@ export async function updateGasFees(request: UpdateGasFeesRequest) {
 }
 
 function getMaxFeePerGas(request: GetGasFeeRequest): string | undefined {
-  const { eip1559, initialParams, suggestedGasFees } = request;
+  const {
+    advancedGasFees,
+    eip1559,
+    initialParams,
+    shouldReadAdvancedGasFees,
+    suggestedGasFees,
+  } = request;
 
   if (!eip1559) {
     return undefined;
   }
 
+  if (shouldReadAdvancedGasFees) {
+    const maxFeePerGas = gweiDecimalToWeiHex(
+      advancedGasFees?.maxBaseFee as string,
```

Comment:
This cast shouldn't be needed if we're instead doing a `if(savedGasFees)`.

Reply:
True that, just quick checked and it's not possible to set undefined in the UI, probably not in the controller too, fixed